tests/dateSelector.spec.ts

```diff
@@ -0,0 +1,140 @@
+import { test, expect } from '@playwright/test';
+import exp from 'constants';
+
+//1. Select the OWNERS menu item in the navigation bar and then select "Search" from the drop-down menu
+test.beforeEach(async ({ page }) => {
+    await page.goto('/')
+    await page.getByText('Owners').click()
+    await page.getByText('Search').click()
+})
+
+test('select the desired date in the calendar', async ({ page }) => {
+    //In the list of the Owners, locate the owner by the name "Harold Davis" and select this owner
+    await page.getByRole('link', { name: 'Harold Davis' }).click()
+    //3. On the Owner Information page, select "Add New Pet" button
+    await page.getByRole('button', { name: 'Add New Pet' }).click()
+    //4. In the Name field, type any new pet name, for example "Tom"
+    const icon = page.locator('span.glyphicon.form-control-feedback').first()
+    //assert the icon 'x'
+    await expect(icon).toHaveClass(/glyphicon-remove/)
+    await page.locator('#name').fill('Tom')
+    //5. Add the assertion of icon in the input field the "checkmark" icon
+    await expect(icon).toHaveClass(/glyphicon-ok/)
+    //6. Click on the calendar icon for the "Birth Date" field
+    await page.getByLabel('Open calendar').click()
+    //7. Using calendar selector, select the date "May 2nd, 2014"
+    const birthYear = '2014';
+    const birthMonth = '05';
+    const birthDay = '02';
+    const dateToAssert = `${birthYear}/${birthMonth}/${birthDay}`
+
+    await page.getByLabel('Choose month and year').click()
+    await page.getByLabel('Previous 24 years').click()
+    await page.getByRole('button', { name: birthYear }).click()
+    await page.getByLabel(`${birthMonth} ${birthYear}`).click()
+    await page.getByLabel(dateToAssert).click()
+    await expect(page.locator('[name="birthDate"]')).toHaveValue(dateToAssert)
+    //9. Select the type of pet "dog" and click "Save Pet" button
+    await page.getByLabel('Type').selectOption('dog')
+    await page.getByRole('button', { name: 'Save Pet' }).click()
+    //10. On the Owner Information page, add assertions for the newly created pet. Name is Tom, Birth Date is in the format "2014-05-02", Type is dog
+    const petField = page.locator('td', { has: page.getByText('Tom') })
+    await expect(petField).toContainText(`${birthYear}-${birthMonth}-${birthDay}`)
+    await expect(petField).toContainText('dog')
```

Comment:
This locator for assertion is incorrect.
It validates, that the box, where "Tom" is located has values of date and "dog". But here need to validate that Name is Tom but Birthday is expected date, as well as type is "dog". If developer will make a mistake in data mapping and make like:
Name: 2014-05-02
Type: Tom
The test will pass.
![image](https://github.com/user-attachments/assets/6a6893ea-9544-46cd-a41e-f9bef4eb4522)